picard/util/__init__.py

```diff
@@ -1183,4 +1184,10 @@ def detect_unicode_encoding(path):
         for bom, encoding in ENCODING_BOMS.items():
             if first_bytes.startswith(bom):
                 return encoding
-        return 'utf-8'
+        rawdata = f.read()
```

Comment:
It could be limited to a certain amount of bytes to avoid accidentally loading excessively huge files. Not sure how many, though. If you read not enough you risk reading a file with wrong encoding. Imagine a iso-8859-1 encoded file where all is ASCII but the last track title contains an french accent. If you miss reading the last track the guessed encoding could be ASCII or UTF-8, both resulting in problems reading the last title.

So I would limit reading to a byte count where we can assume a typical disc ID log file fits in. The largest log in our test data is currently 11kb. If we limit reading to a couple of MB maybe we would be safe that we get the data, but still small enough that there is no risk of extreme delays in reading the file or even causing major issue like full RAM and swapping.

Reply:
We could limit size to something like 256kb, as it is very unlikely a log file is ever bigger than this.
But if we go this way, we need to pass this limit (as optional, with default) to the method, since this function is not specific to log files (even though it is only used for those atm).

Reply:
Sounds good let's go with 256kb as default, but make it a parameter to `detect_file_encoding`